device-connectors/src/testflinger_device_connectors/devices/zapper_kvm/__init__.py

```diff
@@ -36,15 +39,36 @@ class DeviceConnector(ZapperConnector):
 
     PROVISION_METHOD = "ProvisioningKVM"
 
+    def _validate_user_data(self, encoded_user_data: str):
+        """
+        Assert `base_user_data` argument is a valid base64 encoded YAML.
+        """
+        try:
+            user_data = base64.b64decode(encoded_user_data.encode()).decode()
+            yaml.safe_load(user_data)
+        except (binascii.Error, ValueError) as exc:
+            raise ProvisioningError(
+                "Provided `base_user_data` is not base64 encoded."
+            ) from exc
+        except yaml.YAMLError as exc:
+            raise ProvisioningError(
+                "Provided `base_user_data` is not a valid YAML."
+            ) from exc
+
     def _get_autoinstall_conf(self) -> Optional[Dict[str, Any]]:
         """Prepare autoinstall-related configuration."""
         provision = self.job_data["provision_data"]
 
         if "storage_layout" not in provision:
             return None
```

Comment:
Just to confirm, if no storage_layout is specified, then this means that autoinstall is not supported? If that's not the case, then I think this check needs fixing. Otherwise, it might be nice to make that more clear in the documentation. For example, is there some default storage layout you can accept and then still be able to specify params like `oem`?

Reply:
You're perfectly right, it doesn't make much sense to rely on that single argument. In https://github.com/canonical/testflinger/pull/324/commits/581686fed35a1dbfda9c9a38a0534ebd3c0d2550 I made it more generic: any `autoinstall`-related argument is taken in consideration and possibly validated. Yes, a couple of arguments name have changed, but I think it makes them clearer. There are no job on jenkins yet, it shouldn't be a problem having this breaking change in naming.